code/modules/reagents/oldchem/reagents/reagents_admin.dm

```diff
@@ -51,4 +52,24 @@
 /datum/reagent/adminordrazine/nanites
 	name = "Nanites"
 	id = "nanites"
-	description = "Nanomachines that aid in rapid cellular regeneration."
\ No newline at end of file
+	description = "Nanomachines that aid in rapid cellular regeneration."
+
+
+// For random item spawning. Takes a list of paths, and returns the same list without anything that contains admin only reagents
+
+/proc/adminReagentCheck(var/list/incoming)
+	var/list/outgoing[0]
+	for(var/tocheck in incoming)
+		if(ispath(tocheck))
+			var/check = new tocheck
+			if (istype(check, /atom))
+				var/atom/reagentCheck = check
+				var/datum/reagents/reagents = reagentCheck.reagents
+				var/admin = 0
+				for(var/datum/reagent/reagent in reagents.reagent_list)
```

Comment:
generally you want to be as sane as possible--that said for `for` loops---you want to avoid typecasting whenever possible---but if you're not sure if a list contains only one type, then, by all means, typecast it.

it's just that for the reagents list? It's not possible for anything other than a reagent to be in it, so there's no point it performing the istype check on everything.